#VRML_SIM R2020a utf8
WorldInfo {
  basicTimeStep 16
}
StartZone {
  translation -1.46673 8.52651e-14 1.4733
  rotation 0 1 0 3.14159
  colour 0 0.5 0
}
StartZone {
  translation 1.4727 0 -1.44085
  colour 0 0 0.5
  name "start zone(1)"
}
Viewpoint {
  orientation 0.3562430552903888 0.6979825390747986 0.6212175631001318 3.9030616642406315
  position -0.8490359671131257 6.252127110046091 -0.48146271709043376
  followType "Pan and Tilt Shot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 3.66 3.66
  tileSize 3.66 3.66
  appearance PBRAppearance {
    baseColor 0.9 0.9 0.9
    baseColorMap ImageTexture {
      url [
        "test_arena_B-textures/test_arena_B-search_and_beaconing.png"
      ]
      repeatS FALSE
      repeatT FALSE
    }
    roughness 1
    metalness 0
  }
}
StartZone {
  translation -0.872974 0 -1.46253
  rotation 0 1 0 -1.5707953071795862
  colour 0 0.5 0.5
  name "start zone(2)"
}
DEF ARENA_WALLS Group {
  children [
    Solid {
      translation 1.79 0.15 0.04
      children [
        DEF ARENA_WALL_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0.701961 0.701961 0.701961
            roughness 1
          }
          geometry DEF ARENA_WALL_GEOM Box {
            size 0.08 0.3 3.58
          }
        }
      ]
      name "wall1"
      boundingObject USE ARENA_WALL_GEOM
    }
    Solid {
      translation -0.04 0.15 1.79
      rotation 0 1 0 1.5707996938995747
      children [
        USE ARENA_WALL_SHAPE
      ]
      name "wall2"
      boundingObject USE ARENA_WALL_GEOM
    }
    Solid {
      translation -1.79 0.15 -0.04
      children [
        USE ARENA_WALL_SHAPE
      ]
      name "wall3"
      boundingObject USE ARENA_WALL_GEOM
    }
    Solid {
      translation 0.04 0.15 -1.79
      rotation 0 1 0 -1.5707996938995747
      children [
        USE ARENA_WALL_SHAPE
      ]
      name "wall4"
      boundingObject USE ARENA_WALL_GEOM
    }
  ]
}
DEF MAZE_WALLS Group {
  children [
    MazeWall {
      translation -1.357 0.15 0.779
      rotation 0 -1 0 -5.307179586466759e-06
      name "maze wall custom(30)"
      length 0.75
    }
    MazeWall {
      translation -0.401 0.15 -0.574
      rotation 0 -1 0 -5.307179586466759e-06
      name "maze wall custom(1)"
      length 0.95
    }
    MazeWall {
      translation 0.467 0.15 -0.276
      rotation 0 -1 0 0.6544936938995747
      name "maze wall custom(2)"
      length 0.95
    }
    MazeWall {
      translation 0.848 0.15 0.302
      rotation 0 -1 0 1.5707896938995747
      name "maze wall custom(3)"
      length 0.55
    }
    DEF ZONE_A_WALLS Group {
    }
    DEF ZONE_B_WALLS Group {
    }
    DEF ZONE_C_WALLS Group {
    }
  ]
}
DEF BEACONS Group {
  children [
    DEF RED_CYL Solid {
      translation -0.0759672 0.2 -0.285755
      children [
        DEF BCN_RED Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.4
            radius 0.15
          }
        }
      ]
      boundingObject USE BCN_RED
      physics DEF BCN_PHYS Physics {
        density -1
        mass 0.5
      }
      recognitionColors [
        1 0 0
      ]
    }
    DEF FCA_CYL Solid {
      translation -0.356726 0.1 -1.51029
      children [
        DEF BCN_FCA Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.2
          }
        }
      ]
      name "solid(2)"
      boundingObject USE BCN_FCA
      physics USE BCN_PHYS
      recognitionColors [
        1 0 1
      ]
    }
    DEF TEAL_BOX Solid {
      translation 1.45023 0.2 1.47189
      children [
        DEF BCN_TEAL Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 0.5
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.45 0.4 0.45
          }
        }
      ]
      name "solid(1)"
      boundingObject USE BCN_TEAL
      physics USE BCN_PHYS
      recognitionColors [
        0 0.5 0.5
      ]
    }
    DEF GRN_RECT Solid {
      translation 0.701167 0.15 -0.375253
      rotation 0 1 0 -0.6544996938995746
      children [
        DEF BCN_GRN Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.6 0.3 0.3
          }
        }
      ]
      name "solid(3)"
      boundingObject USE BCN_GRN
      physics USE BCN_PHYS
      recognitionColors [
        0 0.5 0
      ]
    }
    DEF BLU_RECT Solid {
      translation -1.42411 0.11 0.336374
      rotation 0 1 0 1.0471996938995747
      children [
        DEF BCN_BLU Shape {
          appearance PBRAppearance {
            baseColor 0 0 0.5
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.6 0.22 0.25
          }
        }
      ]
      name "solid(4)"
      boundingObject USE BCN_BLU
      physics USE BCN_PHYS
      recognitionColors [
        0 0 0.5
      ]
    }
  ]
}
DEF KAZ Robot {
  translation -0.880656 0.08 -1.46749
  rotation 0 1 0 -1.5707953071795862
  children [
    DistanceSensor {
      translation 0 0.04 0.08
      rotation 0 1 0 -1.5707953071795862
      children [
        DEF dist_s Shape {
          appearance PBRAppearance {
            baseColor 1 0.999969 0.999985
            metalness 0.5
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_front"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.0207056 0.04 0.0772743
      rotation 0 1 0 -1.3089969389957472
      children [
        USE dist_s
      ]
      name "ds_FL15"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.04 0.04 0.0692823
      rotation 0 1 0 -1.0471975511965976
      children [
        USE dist_s
      ]
      name "ds_FL30"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.0565687 0.04 0.0565687
      rotation 0 1 0 -0.7853981633974483
      children [
        USE dist_s
      ]
      name "ds_FL45"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.0692823 0.04 0.04
      rotation 0 1 0 -0.5235987755982988
      children [
        USE dist_s
      ]
      name "ds_FL60"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.0207056 0.04 0.0772741
      rotation 0 1 0 -1.8325957145940461
      children [
        USE dist_s
      ]
      name "ds_FR15"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.0400001 0.04 0.0692823
      rotation 0 1 0 -2.0943951023931953
      children [
        USE dist_s
      ]
      name "ds_FR30"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.0565687 0.04 0.0565687
      rotation 0 1 0 -2.356194490192345
      children [
        USE dist_s
      ]
      name "ds_FR45"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.0692823 0.04 0.04
      rotation 0 1 0 -2.6179938779914944
      children [
        USE dist_s
      ]
      name "ds_FR60"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.0692823 0.04 -0.04
      rotation 0 1 0 -3.6651914291880923
      children [
        USE dist_s
      ]
      name "ds_BR60"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.0692823 0.04 -0.04
      rotation 0 1 0 -5.759586531581287
      children [
        USE dist_s
      ]
      name "ds_BL60"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0 0.04 -0.08
      rotation 0 1 0 1.5708
      children [
        USE dist_s
      ]
      name "ds_back"
      lookupTable [
        0 0 0
        0.05 5 0.1
        0.1 10 0.1
        0.2 20 0.1
        0.3 30 0.1
        0.5 50 0.1
      ]
      type "sonar"
    }
    Transform {
      translation 1.59215e-07 0.07 0.06
      rotation 0 1 0 -5.307179586466759e-06
      children [
        Camera {
          translation 0 0.02 0
          rotation 0 1 0 3.14159
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.0895094 0.0895094 0.0895094
                metalness 0.5
              }
              geometry Sphere {
                radius 0.01
                subdivision 2
              }
            }
          ]
        }
        Shape {
          appearance PBRAppearance {
            baseColor 1 0.999969 0.999985
            metalness 0.5
          }
          geometry Cylinder {
            height 0.03
            radius 0.005
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.08 -0.035 0
      }
      device [
        RotationalMotor {
          name "left_wheel"
        }
      ]
      endPoint Solid {
        translation 0.08 -0.035 0
        rotation 3.794223209220402e-09 -3.794237653310065e-09 1 1.5707999999999995
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.0209354 0.248005 0.772824
              metalness 0.5
            }
            geometry Cylinder {
              height 0.01
              radius 0.04
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_ph Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.08 -0.035 0
      }
      device [
        RotationalMotor {
          name "right_wheel"
        }
      ]
      endPoint Solid {
        translation -0.08 -0.035 0
        rotation 3.815519203243388e-09 -3.81553310027069e-09 1 1.570799999999999
        children [
          USE wheel
        ]
        boundingObject USE wheel
        physics USE wheel_ph
      }
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0.999969 0.999985
        baseColorMap ImageTexture {
          url [
            "textures/checkered_marble.jpg"
          ]
        }
        metalness 0
      }
      geometry Cylinder {
        height 0.12
        radius 0.075
      }
    }
    Solid {
      translation 0 -0.055 -0.08
      scale 0.2 0.2 0.2
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Capsule {
            height 0.1
            radius 0.05
          }
        }
      ]
      name "solid(1)"
      boundingObject Capsule {
        height 0.1
        radius 0.05
        subdivision 16
      }
      physics Physics {
      }
    }
    Solid {
      translation 0 -0.055 0.08
      scale 0.2 0.2 0.2
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Capsule {
            height 0.1
            radius 0.05
          }
        }
      ]
      name "solid(3)"
      boundingObject Capsule {
        height 0.1
        radius 0.05
        subdivision 16
      }
      physics Physics {
      }
    }
  ]
  name "KAZ"
  boundingObject USE BODY
  physics Physics {
  }
  controller "obstacles_avoidance"
}
